File: frontend/src/main/typescript/backend/dto/request/addReleaseEventRequest.ts
import { ReleaseEvent } from "@/backend/dto/lowerLevelStruct";

export interface AddReleaseEventRequest {
  entryId: number;
  event: ReleaseEvent;
}

export interface MassAddReleaseEventRequest {
  subRequests: AddReleaseEventRequest[];
  clientType: string;
}

File: backend/src/main/kotlin/niconicotagger/dto/api/request/AddReleaseEventRequest.kt
package niconicotagger.dto.api.request

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.Valid
import niconicotagger.dto.api.misc.ClientType
import niconicotagger.dto.api.misc.ReleaseEvent

@JsonIgnoreProperties(ignoreUnknown = false)
data class MassAddReleaseEventRequest(
    @field:Valid
    override val subRequests: List<AddReleaseEventRequest>,
    override val clientType: ClientType
) : MassUpdateRequest<AddReleaseEventRequest>

data class AddReleaseEventRequest(
    override val entryId: Long,
    @field:Valid
    override val event: ReleaseEvent
) : ReleaseEventAdditionRequest
